SYS_ATL/prec_analysis.py

```diff
@@ -101,19 +101,20 @@ def map_s(self, s):
             assert ltyp != T.err and ltyp != T.R
 
             # update the type annotation here if needed
-            result[0].type = ltyp
+            result[0] = result[0].update(type=ltyp)
```

Comment:
ADT nodes are now immutable (and this is checked at runtime). Use `.update()` to create a new node with a subset of fields changed.